samcli/hook_packages/terraform/hooks/prepare/resource_linking.py

```diff
@@ -306,7 +361,10 @@ def _process_reference_resource_value(
                 )
 
             tf_dest_res_name = resolved_destination_resource.value[
-                len(self._resource_pair.terraform_resource_type_prefix) : -len(".arn")
+                len(self._resource_pair.terraform_resource_type_prefix) : -len(
```

Comment:
lets add a note on why, otherwise this looks like magic.

Reply:
done